pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -132,9 +133,13 @@ func GetPodConditionFromList(conditions []v1.PodCondition, conditionType v1.PodC
 }
 
 func (pc *PodCache) labelFilter(old, cur *v1.Pod) bool {
-	// If labels updated, trigger proxy push
-	if cur.Status.PodIP != "" && !maps.Equal(old.Labels, cur.Labels) {
-		pc.proxyUpdates(cur.Status.PodIP)
+	// If labels/annotations updated, trigger proxy push
+	labelsChanged := !maps.Equal(old.Labels, cur.Labels)
+	// Annotations are only used in endpoints in one case, so just compare that one
+	relevantAnnotationsChanged := old.Annotations[constants.AmbientRedirection] != cur.Annotations[constants.AmbientRedirection]
```

Comment:
I donot think we should change the behavior of redirect behavior on flight

Reply:
In envoy or in all of ambient? 

If we do it in ztunnel but not envoy it causes an outage when you switch

Reply:
all ambient, remember we have some fixed ops based on labels for ambient in CNI.

Reply:
You can change the control plane, but you cannot make the CNI to rework

Reply:
This topic has been thoroughly discussed already and the conclusion was that dynamic changes was a hard requirement for ambient. See https://github.com/istio/istio/issues/48876 for some of the past discussion.